services/venice-router/src/main/java/com/linkedin/venice/router/VeniceRouterConfig.java

```diff
@@ -247,7 +247,7 @@ private void checkProperties(VeniceProperties props) {
     smartLongTailRetryEnabled = props.getBoolean(ROUTER_SMART_LONG_TAIL_RETRY_ENABLED, true);
     smartLongTailRetryAbortThresholdMs = props.getInt(ROUTER_SMART_LONG_TAIL_RETRY_ABORT_THRESHOLD_MS, 100);
     // Default: -1 means this feature is not enabled.
-    longTailRetryMaxRouteForMultiKeyReq = props.getInt(ROUTER_LONG_TAIL_RETRY_MAX_ROUTE_FOR_MULTI_KEYS_REQ, -1);
+    longTailRetryMaxRouteForMultiKeyReq = props.getInt(ROUTER_LONG_TAIL_RETRY_MAX_ROUTE_FOR_MULTI_KEYS_REQ, 2);
```

Comment:
I wonder if it would be better to set the limit to a higher number, such as 5.

Going from unlimited retries to 2 retries feels a bit forceful. But of course, it all depends on whether there are requests that would've typically resolved after 2+ retries that we would now see producing errors, which I have no idea about. We definitely don't want to encounter another, so maybe 2 is the right number here.

Reply:
We always can override these configs for setups which serves in-memory and support high batch key count. For others, this config being 1 or 2 is good as anything more might be counterproductive (i,e retries are to help the long tails and thats supposed to help with servers that are busy or in bad state, keep retrying in those scenarios are going to put the system in a bad state).

Reply:
yes, with higher retry limit for a large batch size they servers are going to pounded even more and causing further retry storm.